project_template/webpack.config.js

```diff
@@ -29,31 +29,26 @@ const plugins = [
     minify: false
   }),
   new Clean(["dist"], { verbose: false, exclude: [".keep"] }),
-  new webpack.optimize.CommonsChunkPlugin({
-    name: "vendor"
-  }),
   extractShellCss,
   extractOtherCss,
   new SpriteLoaderPlugin()
 ];
 
-if (isProd) {
-  plugins.push(
-    new webpack.LoaderOptionsPlugin({ minimize: true, debug: false }),
-    new webpack.optimize.UglifyJsPlugin(uglify)
-  );
-} else {
-  if (hotReload) {
-    plugins.push(
-      // enable HMR globally
-      new webpack.HotModuleReplacementPlugin()
-    );
-  }
+const optimization = {
+  splitChunks: {
+    name: "vendor"
```

Comment:
@pascalw in my experience using the default webpack4 heuristics produces better results. This is what I get with the setup in this PR:

<img width="562" alt="screen shot 2018-05-10 at 17 52 44" src="https://user-images.githubusercontent.com/234036/39882362-72e31062-547b-11e8-8e2f-e22e344d8eaf.png">

It seems like all the vendor stuff is still in the main `app.js`, and a separate chunk is created for the `vendors` entrypoint which ends up having duplicate stuff.

Running _without_ the vendors entrypoint and with the following optimization produces much better results:

```javascript
  splitChunks: {
    chunks: "all"
  }
```

<img width="599" alt="screen shot 2018-05-10 at 17 52 59" src="https://user-images.githubusercontent.com/234036/39882454-c022474e-547b-11e8-9bb3-e74d6913a351.png">

As far as I can tell webpack 4 already expects to produce a vendors chunk per entrypoint (or multiple, depending on the size of the dependencies).

Reply:
That's cool! I'll update this PR to let Webpack do it's job :-)